rdt/transformers/text.py

```diff
@@ -211,14 +211,18 @@ def _reverse_transform(self, data):
             self.reset_randomization()
             remaining = self.generator_size
 
-        if remaining >= sample_size:
-            reverse_transformed = [next(self.generator) for _ in range(sample_size)]
-            self.generated += sample_size
-
-        else:
-            generated_values = list(self.generator)
-            reverse_transformed = generated_values[:]
-            self.generated = self.generator_size
+        generated_values = []
+        while len(generated_values) < sample_size:
+            try:
+                generated_values.append(next(self.generator))
+                self.generated += 1
+            except Exception:  # pylint: disable=broad-exception-caught
+                # Can't generate more rows without collision so breaking out of loop
+                break
```

Comment:
This will not raise an error, would that affect the current functionality when attempting to generate `unique` primary keys ?

Reply:
I'm a little confused, the current functionality doesn't seem to do anything with errors. Is there something in SDV you're referring to?

This idea here is to sample as much as possible until you have enough or a `StopIteration` is raised because the generator is out of values. I can change it to catch specifically that type of error